### PR TITLE
Rename the `object_struct` field labels and constants

### LINKED ISSUE #990: `object_struct` wram labels should match their offset constants
Currently the wram labels for the `object_struct` have things such as:

```
\1StandingMapX::      db
\1StandingMapY::      db
\1LastMapX::          db
\1LastMapY::          db
```

However the offset constants use:

```
DEF OBJECT_NEXT_MAP_X          rb ; 10
DEF OBJECT_NEXT_MAP_Y          rb ; 11
DEF OBJECT_MAP_X               rb ; 12
DEF OBJECT_MAP_Y               rb ; 13
```

In the above examples, the wram labels should be renamed to match the offset constants. There are a few others here than need addressed.

Also we should investigate the true purpose of the following, and come up with better names for these:

```
DEF OBJECT_1C                  rb ; 1c
DEF OBJECT_1D                  rb ; 1d
DEF OBJECT_1E                  rb ; 1e
DEF OBJECT_1F                  rb ; 1f
```

### PR DESCRIPTION
**Pull request by Rangi42**
Fixes #990

### CODE REVIEW

File: constants/map_object_constants.asm [hunk 1]
```diff
@@ -31,13 +31,13 @@ DEF OBJECT_MOVEMENT_BYTE_INDEX rb ; 1b
 DEF OBJECT_1C                  rb ; 1c
```
Comment (vulcandth) [inline constants/map_object_constants.asm:31]:
```suggestion
DEF OBJECT_ANON_JUMPTABLE_INDEX                  rb ; 1c
```
You'll have to fix the formatting and the associated wram label, and everywhere it's used.

File: constants/map_object_constants.asm [hunk 2]
```diff
@@ -31,13 +31,13 @@ DEF OBJECT_MOVEMENT_BYTE_INDEX rb ; 1b
...
 DEF OBJECT_1D                  rb ; 1d
 DEF OBJECT_1E                  rb ; 1e
```
Comment (vulcandth) [inline constants/map_object_constants.asm:33]: These seem to be used for a couple purposes... One of which seems to store the address of the start of another object used in `StepFunction_TrackingObject`. So it's fine they remain as they are now.. just wanted to mention it.

**Review by vulcandth — CHANGES_REQUESTED** [review]
After that one requested change; I think this is good to merge. Thanks 👍

File: constants/map_object_constants.asm (hunk 1 above)
Comment (vulcandth) [reply constants/map_object_constants.asm:31]: (If you agree with this, I can push the required changes to the PR for you. Since I know you are busy.)

**Review by vulcandth — APPROVED** [review]

**Review by vulcandth — APPROVED** [review]

**Comment by mid-kid** [comment]
`CMDQUEUE_ANON_JUMPTABLE_INDEX` -> `CMDQUEUE_JUMPTABLE_INDEX` honestly. The only thing that makes it "anonymous" is an implementation detail.

**Review by vulcandth — APPROVED** [review]

**Review by mid-kid — APPROVED** [review]

**Pull request merged**